assets/dev/js/editor/container/model/children-collection.js

```diff
@@ -0,0 +1,75 @@
+export default class ChildrenCollection extends Array {
```

Comment:
Make it a general util.
```suggestion
export default class RecursiveArray extends Array {
```

Reply:
It is not array generic the class know about the existence of `.children`.

Reply:
And now?